or_benchmarks/src/CheckerResultModule.cpp

```diff
@@ -0,0 +1,243 @@
+    {
+        OpenRAVE::KinBody::LinkConstPtr plink1 = DeserializeLink("link1",check_log_val);
+        OpenRAVE::KinBody::LinkConstPtr plink2 = DeserializeLink("link2",check_log_val);
+
+        start = std::chrono::high_resolution_clock::now();
+        check_result = checker_ptr -> CheckCollision(plink1, plink2);
+        end = std::chrono::high_resolution_clock::now();
+    }
+
+    else if (methodname == "CheckCollision_link_body")
+    {
+        OpenRAVE::KinBody::LinkConstPtr plink = DeserializeLink("link",check_log_val);
+        OpenRAVE::KinBodyConstPtr pbody = DeserializeKinBody("body",check_log_val);
+
+        start = std::chrono::high_resolution_clock::now();
+        check_result = checker_ptr -> CheckCollision(plink, pbody);
+        end = std::chrono::high_resolution_clock::now();
+    }
+
+    else if (methodname == "CheckCollision_link_with_exclusions")
+    {
+        OpenRAVE::KinBody::LinkConstPtr plink = DeserializeLink("link",check_log_val);
+
+        std::vector<OpenRAVE::KinBodyConstPtr> vbodiesexcluded = DeserializeKinBodyList("bodies_excluded",check_log_val);
+        std::vector<OpenRAVE::KinBody::LinkConstPtr> vlinksexcluded = DeserializeLinkList("links_excluded",check_log_val);
+
+        start = std::chrono::high_resolution_clock::now();
+        check_result = checker_ptr -> CheckCollision(plink, vbodiesexcluded, vlinksexcluded);
+        end = std::chrono::high_resolution_clock::now();
+    }
+
+    else if (methodname == "CheckCollision_body_with_exclusions")
+    {
+        OpenRAVE::KinBodyConstPtr pbody = DeserializeKinBody("body",check_log_val);
+
+        std::vector<OpenRAVE::KinBodyConstPtr> vbodiesexcluded = DeserializeKinBodyList("bodies_excluded",check_log_val);
+        std::vector<OpenRAVE::KinBody::LinkConstPtr> vlinksexcluded = DeserializeLinkList("links_excluded",check_log_val);
+
+        start = std::chrono::high_resolution_clock::now();
+        check_result = checker_ptr -> CheckCollision(pbody, vbodiesexcluded, vlinksexcluded);
+        end = std::chrono::high_resolution_clock::now();
+    }
+
+    else if (methodname == "CheckStandaloneSelfCollision")
+    {
+        OpenRAVE::KinBodyConstPtr pbody = DeserializeKinBody("body",check_log_val);
+
+        start = std::chrono::high_resolution_clock::now();
+        check_result = checker_ptr -> CheckStandaloneSelfCollision(pbody);
+        end = std::chrono::high_resolution_clock::now();
+    }
+
+    else if (methodname == "CheckStandaloneSelfCollisionBaked")
+    {
+        OpenRAVE::KinBodyConstPtr pbody = DeserializeKinBody("body",check_log_val);
+
+        start = std::chrono::high_resolution_clock::now();
+        check_result = checker_ptr -> CheckStandaloneSelfCollision(pbody);
+        end = std::chrono::high_resolution_clock::now();
+    }
+
+    else
+    {
+        throw std::runtime_error("Unknown method - "+methodname);
+    }
+
+    std::chrono::duration<double> elapsed_seconds = end-start;
```

Comment:
Hmm it did work I think because it is seconds by default but I am casting for good measure